### PR TITLE
Merge rustup's crates

### LINKED ISSUE #1644: Merge some of the crates (probably only rustup and rustup-dist)
Split out from https://github.com/rust-lang/rustup.rs/issues/1611

The current crates tree is:

```
rustup-win-installer
└── rustup
    └── rustup-dist
        └── rustup-utils
            └── download

[dev-dependencies]
└── rustup-mock
```

WDYT about squashing that down to:

```
rustup-win-installer
└── rustup
```

And the only reason to keep those separate is because I can't run rustup-win-installer with its build.rs.  WDYT?

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #1644

### CODE REVIEW

**Comment by dwijnand** [comment]
@nrc / @kinnison let me know if there's anything I can do to make it easier to review this (open to suggestions).  I can present this in a call, if running through it together helps (I've had success with this strategy, as both a submitter and reviewer).

File: ci/run.sh [hunk 1]
```diff
@@ -11,7 +11,6 @@ rustfmt -vV
 cargo build --locked -v --release --target "$TARGET" --features vendored-openssl
 
 if [ -z "$SKIP_TESTS" ]; then
-  cargo test --release -p download --target "$TARGET" --features vendored-openssl
```
Comment (kinnison) [inline ci/run.sh]: Why remove the tests for the download crate?

File: tests/cli-exact.rs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 //! Yet more cli test cases. These are testing that the output
 //! is exactly as expected.
 
-use rustup_mock::clitools::{
+pub mod mock;
```
Comment (kinnison) [inline tests/cli-exact.rs:4]: Does this approach mean we end up spending CPU compiling mock over and over?

I'm not entirely upset if the answer is yes, but if so, could it be made a `#[cfg(test)] pub mod mock` in src/ instead?  Would that work?

**Review by kinnison — APPROVED** [review]
Pretty much spot on what I was expecting.

Two comments, but if they don't end up implying changes, then :+1:

File: ci/run.sh (hunk 1 above)
Comment (dwijnand) [reply ci/run.sh]: My mistake

File: tests/cli-exact.rs (hunk 2 above)
Comment (dwijnand) [reply tests/cli-exact.rs:4]: Yeah we do.  In cargo this is fixed by making one "testsuite" binary, with all tests.

I'll try the cfg test approach.

Comment (dwijnand) [reply tests/cli-exact.rs:4]: Doesn't work.

**Review by kinnison — APPROVED** [review]

**Pull request merged**